frontend/src/metabase/actions/containers/ActionCreator/FormCreator/utils.ts

```diff
@@ -60,6 +60,9 @@ export const reorderFields = (
 };
 
 export const hasNewParams = (
-  params: Parameter[],
+  parameters: Parameter[],
   formSettings: ActionFormSettings,
-) => !!params.find(param => !formSettings.fields[param.id]);
+) => {
+  const fieldIds = Object.keys(formSettings.fields || {});
+  return parameters.some(parameter => !fieldIds.includes(parameter.id));
```

Comment:
nit: Although it might look easier to understand, I think it is a bit indirect. I don't like the fact that we have a dictionary, converts it to an array, then loops over it. Kind of defeat the purpose of having a dictionary in the first place.

```suggestion
  return parameters.some(parameter => !(formSettings.fields || {})[parameter.id]);
```